[Build] Disable WTF/bmalloc modularization in some SDK versions

emw-apple opened the pull request:
#### 289beb67df00adebab914643d66419c3a5511253
<pre>
[Build] Disable WTF/bmalloc modularization in some SDK versions
<a href="https://bugs.webkit.org/show_bug.cgi?id=299210">https://bugs.webkit.org/show_bug.cgi?id=299210</a>
<a href="https://rdar.apple.com/157498429">rdar://157498429</a>

Reviewed by Jonathan Bedard.

Some internal builds are affected by a bug in the SDK &lt;<a href="https://rdar.apple.com/157124281">rdar://157124281</a>&gt;
which prevents us from integrating WTF and bmalloc modularization.
Disable modulariztaion for affected SDK versions.

The bug only occurs in install-style builds, so keep modularization
enabled in at-desk builds to make it easier to detect issues with
modularized code before a change reaches CI.

* Source/WTF/Configurations/WTF.xcconfig: As a drive-by cleanup, change
  from &quot;NO&quot; to undefined when modularization is disabled. This is the
  convention we use in other build settings, and it keeps the setting
  from showing up in build logs.
* Source/WTF/WTF.xcodeproj/project.pbxproj: Add quoting to some
  environment variables in shell script phase.
* Source/WebKit/Configurations/WebKit.xcconfig: Swift-C++ interop
  requires modularized WTF, so restrict it in the same manner.
* Source/bmalloc/Configurations/bmalloc.xcconfig: Ditto WTF.xcconfig.

Canonical link: <a href="https://commits.webkit.org/300431@main">https://commits.webkit.org/300431@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5897a8703c1569385c33d1430fa6edb99148c6a5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122518 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/42226 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32911 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/129122 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/74617 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42944 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/50819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/129122 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/74617 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125470 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/34256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/109703 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/129122 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/33241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/27859 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/72609 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/114666 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/103908 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/28070 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/131850 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/121044 "Built successfully and passed tests") | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/49459 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/50819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/131850 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49834 "Build is in progress. Recent messages:") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/105923 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/131850 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/25059 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/46228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19363 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/49317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/55066 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/151203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/48784 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/151203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/52135 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/50466 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

adetaylor commented on Source/WTF/Configurations/WTF.xcconfig:
```diff
@@ -52,10 +52,17 @@ WK_HEADERS_VFS_FILE = $(CPP_HEADERMAP_PRODUCT_HEADERS_VFS_FILE);
 GENERATE_WTF_MODULEMAP_IOS_SINCE_26 = YES;
-GENERATE_WTF_MODULEMAP_IOS_BEFORE_26 = NO;
\ No newline at end of file
+GENERATE_WTF_MODULEMAP_IOS_BEFORE_26 = NO;
```
You still have `NO` here

adetaylor commented on Source/bmalloc/Configurations/bmalloc.xcconfig:
```diff
@@ -24,15 +24,22 @@
 
 EXCLUDED_SOURCE_FILE_NAMES = $(inherited) $(EXCLUDED_SOURCE_FILE_NAMES_$(GENERATE_BMALLOC_MODULEMAP));
+EXCLUDED_SOURCE_FILE_NAMES_ = module.modulemap;
 EXCLUDED_SOURCE_FILE_NAMES_NO = module.modulemap;
```
Nit: Maybe you could zap this line?

emw-apple replied:
I kept it around because `$(WK_NOT_$(DEPLOYMENT_LOCATION))` above will expand to "NO" in `install` builds (i.e. DEPLOYMENT_LOCATION=YES). Kind of awkward either way :shrug:

emw-apple committed:
[Build] Disable WTF/bmalloc modularization in some SDK versions
https://bugs.webkit.org/show_bug.cgi?id=299210
rdar://157498429

Reviewed by Jonathan Bedard.

Some internal builds are affected by a bug in the SDK <rdar://157124281>
which prevents us from integrating WTF and bmalloc modularization.
Disable modulariztaion for affected SDK versions.

The bug only occurs in install-style builds, so keep modularization
enabled in at-desk builds to make it easier to detect issues with
modularized code before a change reaches CI.

* Source/WTF/Configurations/WTF.xcconfig: As a drive-by cleanup, change
  from "NO" to undefined when modularization is disabled. This is the
  convention we use in other build settings, and it keeps the setting
  from showing up in build logs.
* Source/WTF/WTF.xcodeproj/project.pbxproj: Add quoting to some
  environment variables in shell script phase.
* Source/WebKit/Configurations/WebKit.xcconfig: Swift-C++ interop
  requires modularized WTF, so restrict it in the same manner.
* Source/bmalloc/Configurations/bmalloc.xcconfig: Ditto WTF.xcconfig.

Canonical link: https://commits.webkit.org/300431@main
